Move flag util pkgs to k8s.io/component-base/cli

Klaven opened the pull request:
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:

This PR is in direct response the the creation of the component-base WG and is work that is described in the backlog. This PR moves the `apiserver/pkg/util/[flag|globalflag]` -> `component-base/cli/[flag|globalflag]`

**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes none# 

**Special notes for your reviewer**:
This PR has 3 commits, The first is code changes, the second is auto gen'ed files and the last is updating lint exceptions


**Does this PR introduce a user-facing change?**:
<!--  
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
none
```

mtaufen commented:
/ok-to-test

Klaven commented:
hmmmmm...... I ran `./hack/verify-godeps.sh` and it passed on my machine. am I missing something?

Klaven commented:
/test pull-kubernetes-godeps

luxas commented on test/e2e_node/services/kubelet.go:
```diff
@@ -31,7 +31,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	utilflag "k8s.io/apiserver/pkg/util/flag"
+	utilflag "k8s.io/component-base/cli/flag"
```
maybe componentflag or something else?

Klaven replied:
componentflag seems long too me, what about `cliflag`?

Klaven replied:
This will also incur much larger code changes through-out instead of just imports.

stewart-yu replied:
IMO, using `componentflag` and `componentglobalflag` in all change

Klaven replied:
after talking with luxas I changed them all to cliflag.

luxas reviewed (commented):
Thanks @Klaven!
This looks like a pretty clean move 😄

Two improvements yet please:
 - We should settle on one import name for `k8s.io/component-base/cli/flag`. Whether that's `componentflag`/`cliflag`/`utilflag` I don't really care, but we should be consistent across the board.
 - the `TerminalSize` function in `sanctioned.go` _should not_ be moved (yet at least). Instead, move that func (only) to `k8s.io/apiserver/pkg/util/term`. This is due to that it pulls in a lot of other godeps, and we want to try to add as little deps as just possible for the moment. And it ain't directly related to flags in any case so :)

thockin commented:
It looks like you have great reviewers.   Assign to me if you need approval.

stewart-yu commented on cmd/cloud-controller-manager/app/controllermanager.go:
```diff
@@ -30,11 +30,11 @@ import (
 	"k8s.io/client-go/tools/leaderelection/resourcelock"
 	cloudprovider "k8s.io/cloud-provider"
+	apiserverflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/cli/globalflag"
```
can we rename 'apiserverflag'?
the same in others place

Klaven replied:
done. it is now `cliflag`. On a side note. I did not name any of the `globalflag` imports as it was never named before and left those imports bare.

stewart-yu commented on cmd/cloud-controller-manager/app/options/options.go:
```diff
@@ -22,18 +22,18 @@ import (
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
keep original better, and others place

Klaven replied:
I'm not sure how this happened. nice catch.

stewart-yu commented on cmd/kube-apiserver/app/options/options_test.go:
```diff
@@ -27,11 +27,11 @@ import (
 	audittruncate "k8s.io/apiserver/plugin/pkg/audit/truncate"
 	restclient "k8s.io/client-go/rest"
+	utilflag "k8s.io/component-base/cli/flag"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
```
rename `utilflag`, ans others place

stewart-yu commented on cmd/kube-apiserver/app/server.go:
```diff
@@ -35,7 +35,7 @@ import (
 	extensionsapiserver "k8s.io/apiextensions-apiserver/pkg/apiserver"
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
```
keep original

luxas replied:
this is autogenerated and does no harm

Klaven replied:
I think this is the last thing. please let me know if I need to go back and update this.

roycaihw commented:
/assign @cheftako

Klaven commented:
will look into bazel-build failure in the morning. I ran `./hack/update-bazel.sh` must have missed running something else.

Klaven commented:
@luxas I have corrected the issues and got all the failing tests working except the `ete-containerized` one and I think that one is ok?

Klaven commented:
I will squash the PR when you all are finished looking over it.

Klaven commented:
/test pull-kubernetes-e2e-gce

luxas commented on staging/src/k8s.io/kube-aggregator/Godeps/Godeps.json:
```diff
@@ -6,14 +6,6 @@
 		"./..."
 	],
 	"Deps": [
-		{
```
Great to cut this unnecessary dependency!

luxas commented on hack/.golint_failures:
```diff
@@ -644,6 +643,7 @@ staging/src/k8s.io/code-generator/cmd/client-gen/types
 staging/src/k8s.io/code-generator/cmd/conversion-gen/generators
 staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf
 staging/src/k8s.io/code-generator/cmd/lister-gen/generators
+staging/src/k8s.io/component-base/cli/flag
```
This is ok, but eventually we should fix this

luxas reviewed (approved):
/lgtm
/approve

Thanks @Klaven! This looks like a clean move of packages :+1:

stewart-yu commented:
/retest

stewart-yu commented on cmd/kube-controller-manager/app/controllermanager.go:
```diff
@@ -34,8 +34,8 @@ import (
 	"k8s.io/apimachinery/pkg/util/term"
+	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/server"
```
i wonder why this change?

stewart-yu commented on staging/src/k8s.io/apiserver/pkg/util/term/term.go:
```diff
@@ -36,4 +36,4 @@ func TerminalSize(w io.Writer) (int, int, error) {
 		return 0, 0, err
 	}
 	return int(winsize.Width), int(winsize.Height), nil
-}
```
why this change in this commit?

stewart-yu commented on cmd/cloud-controller-manager/app/controllermanager.go:
```diff
@@ -26,8 +26,8 @@ import (
 
 	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/term"
```
the same

stewart-yu reviewed (commented):
i think we should make the commit log more graceful:sweat_smile:

rosti commented:
/assign @thockin 
for approval, as suggested by @luxas on Slack.

rosti commented:
/test pull-kubernetes-local-e2e-containerized
/test pull-kubernetes-e2e-kops-aws

thockin reviewed (commented):
/lgtm
/approve

rosti commented:
@Klaven the PR needs rebasing

nikhita commented:
Cancelling lgtm to stop retesting, as this PR needs a rebase.

/lgtm cancel

nikhita commented:
/unassign

Commit:
Moved flag and globalflag

Moved all flag code from `staging/src/k8s.io/apiserver/pkg/util/[flag|globalflag]` to `component-base/cli/[flag|globalflag]` except for the term function because of unwanted dependencies.

Commit:
updated golint_failures to match new project structure.

Commit:
autogen files to support new project structure.

luxas commented:
/lgtm

luxas commented:
/retest
